Tools/TestWebKitAPI/Tests/WebKitCocoa/TextManipulation.mm

```diff
@@ -3852,6 +3853,42 @@ - (void)_webView:(WKWebView *)webView didFindTextManipulationItem:(_WKTextManipu
     EXPECT_WK_STREQ([webView stringByEvaluatingJavaScript:@"document.getElementById('host').shadowRoot.textContent"], "world");
 }
 
+TEST(TextManipulation, CompleteTextManipulationDoesNotFillAutoFilledField)
+{
+    auto delegate = adoptNS([[TextManipulationDelegate alloc] init]);
+    RetainPtr configuration = [WKWebViewConfiguration _test_configurationWithTestPlugInClassName:@"WebProcessPlugInWithInternals" configureJSCForTesting:YES];
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 400, 400) configuration:configuration.get()]);
+    [webView _setTextManipulationDelegate:delegate.get()];
+    [webView synchronouslyLoadHTMLString:@"<!DOCTYPE html><body><input id='textField'></body>"];
+
+    done = false;
+    [webView _startTextManipulationsWithConfiguration:nil completion:^{
+        done = true;
+    }];
+    Util::run(&done);
+
+    Vector<RetainPtr<_WKTextManipulationItem>> items;
+    done = false;
+    [delegate setItemCallback:[&] (_WKTextManipulationItem *item) {
+        items.append(item);
+        done = true;
+    }];
+
+    [webView stringByEvaluatingJavaScript:@"document.getElementById('textField').value = 'foo'"];
+    Util::run(&done);
+
+    [webView stringByEvaluatingJavaScript:@"internals.setAutofilled(document.getElementById('textField'), true)"];
+
+    done = false;
+    __block auto item = createItem([items[0] identifier], { { [items[0] tokens][0].identifier, @"bar" } });
```

Comment:
Does this have to be `__block`?

Reply:
Good catch! The `__block` isn't necessary here.